datanode/api/trading_data_v2.go

```diff
@@ -2534,7 +2534,15 @@ func (t *TradingDataServiceV2) ListPaidLiquidityFees(ctx context.Context, req *v
 		assetID = ptr.From(entities.AssetID(*req.AssetId))
 	}
 
-	stats, pageInfo, err := t.paidLiquidityFeesStatsService.List(ctx, marketID, assetID, req.EpochSeq, req.PartyIds, pagination)
+	partyIDs := req.PartyIds
+	if req.IncludeDerivedParties != nil && *req.IncludeDerivedParties {
+		subKeys, err := t.ammPoolService.GetSubKeysForParties(ctx, partyIDs, marketID)
```

Comment:
Theres already a:
```
func (t *TradingDataServiceV2) getDerivedParties()
```
method that all the other API's with a similar `includeDerivedParties` flag use. It gets the amm-keys in a slightly different way, I don't have an opinion on which is best, but we should use the same function everywhere so we don't end up with inconsistencies between API.

Reply:
Yeah, I saw that method, decided not to use it, because in the way I'd call it here, it'd get all markets, and generate all derived keys for the party/parties (this endpoint is most likely going to be called for a single party anyway). I figured hitting the DB with a single query that may or may not return any derived keys, and then fetching the data for a set of keys that all exist made more sense to me than to use a set of keys that may not exist. That method also returns the data as a map, so FWIW, it just _looked_ a bit messy to iterate over a map just to append to the slice.

Reply:
Makes sense. Can we switch the other API to use your function then?

Reply:
and that is done now